go/oasis-test-runner/scenario/e2e/stake.go

```diff
@@ -0,0 +1,447 @@
+package e2e
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	"github.com/oasislabs/oasis-core/go/common/quantity"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common/flags"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/common/grpc"
+	"github.com/oasislabs/oasis-core/go/oasis-node/cmd/stake"
+	"github.com/oasislabs/oasis-core/go/oasis-test-runner/env"
+	"github.com/oasislabs/oasis-core/go/oasis-test-runner/oasis"
+	"github.com/oasislabs/oasis-core/go/oasis-test-runner/scenario"
+)
+
+const (
+	// Init balance in the genesis block.
+	initBalance = 100_000_000_000
+
+	// Test transfer amount.
+	transferAmount = 1000
+
+	// Test burn amount.
+	burnAmount = 2000
+
+	// Test escrow amount.
+	escrowAmount = 3000
+
+	// Transaction fee amount.
+	feeAmount = 10
+
+	// Transaction fee gas.
+	feeGas = 10
+
+	// Source address in the genesis block.
+	srcAddress = "4ea5328f943ef6f66daaed74cb0e99c3b1c45f76307b425003dbc7cb3638ed35"
+
+	// Test transfer destination address.
+	transferAddress = "5ea5328f943ef6f66daaed74cb0e99c3b1c45f76307b425003dbc7cb3638ed35"
+
+	// Test escrow address.
+	escrowAddress = "6ea5328f943ef6f66daaed74cb0e99c3b1c45f76307b425003dbc7cb3638ed35"
+)
+
+var (
+	// StakeCLI is the staking scenario.
+	StakeCLI scenario.Scenario = &stakeCLIImpl{
+		basicImpl: basicImpl{},
+		logger:    logging.GetLogger("scenario/e2e/stake"),
+	}
+)
+
+type stakeCLIImpl struct {
+	basicImpl
+
+	logger *logging.Logger
+}
+
+func (s *stakeCLIImpl) Name() string {
+	return "stake-cli"
+}
+
+func (s *stakeCLIImpl) Fixture() (*oasis.NetworkFixture, error) {
+	f, err := s.basicImpl.Fixture()
+	if err != nil {
+		return nil, err
+	}
+
+	// We will mock epochs for reclaiming the escrow.
+	f.Network.EpochtimeMock = true
+
+	return f, nil
+}
+
+func (s *stakeCLIImpl) Run(childEnv *env.Env) error {
+	if err := s.net.Start(); err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	logger.Info("waiting for nodes to register")
+	if err := s.net.Controller().WaitNodesRegistered(ctx, 3); err != nil {
+		return errors.Wrap(err, "waiting for nodes to register")
```

Comment:
Use `fmt.Errorf("waiting for nodes to register: %w", err)` instead (everywhere). And remove the external errors import.